components/project-card/project-card.jsx

```diff
@@ -41,33 +43,80 @@ class ProjectCard extends React.Component {
   }
 
   setInitialBookmarkedStatus() {
-    let bookmarks = getBookmarks();
+    if (user.loggedin) {
+      this.setState({
+        bookmarked: this.props.isBookmarked
+      });
+
+      return;
+    }
+
+    let bookmarks = bookmarkManager.bookmarks.get();
     let bookmarked;
-    let index;
 
     if (bookmarks) {
-      index = bookmarks.indexOf(this.props.id);
-      if (index > -1) {
-        bookmarked = true;
-      } else {
-        bookmarked = false;
-      }
+      bookmarked = bookmarks.indexOf(this.props.id) > -1;
       this.setState({bookmarked: bookmarked});
     }
   }
 
-  bookmarkProject(bookmarks) {
-    ReactGA.event(this.createGaEventConfig(`Bookmark button`, `Bookmarked`));
+  bookmarkProjectOnLs(bookmarks) {
```

Comment:
Minification means function names don't matter. It is *much* better to call this `bookmarkToLocalStorage` than using "OnLs". New/future devs will have no idea what LS refers to.